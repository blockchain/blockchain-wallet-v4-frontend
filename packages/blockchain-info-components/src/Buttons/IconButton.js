import React from 'react'
import styled from 'styled-components'

import { Icon } from '../Icons'
import Button from './Button'

const BaseIconButton = styled(Button)`
  width: auto;
  & > :first-child {
    margin-right: 8px;
  }
`

const selectColor = (nature, disabled) => {
  if (disabled) {
    return 'white'
  }

  switch (nature) {
    case 'primary':
    case 'secondary':
    case 'copy':
    case 'received':
    case 'sent':
    case 'transferred':
    case 'logout':
    case 'dark':
      return 'white'
    case 'light':
    case 'white-transparent':
    case 'white-blue':
      return 'blue600'
    default:
      return 'grey800'
  }
}

const IconButton = props => {
  const { children, name, nature, ...rest } = props
  const color = selectColor(nature)

  return (
    <BaseIconButton {...rest}>
      <Icon name={name} color={color} />
      {children}
    </BaseIconButton>
  )
}

export default IconButton
