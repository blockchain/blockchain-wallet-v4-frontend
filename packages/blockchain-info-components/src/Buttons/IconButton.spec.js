import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import IconButton from './IconButton'

describe('IconButton component', () => {
  it('default renders correctly', () => {
    const component = shallow(<IconButton name='test' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
