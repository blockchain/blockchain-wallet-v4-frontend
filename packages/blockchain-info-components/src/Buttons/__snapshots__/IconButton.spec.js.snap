// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component default renders correctly 1`] = `
<IconButton__BaseIconButton>
  <Icon
    color="gray-6"
    cursor={false}
    name="test"
    size="16px"
    weight={400}
  />
</IconButton__BaseIconButton>
`;
